backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/monitoring/VmAnalyzer.java

```diff
@@ -874,15 +874,16 @@ private void proceedDisappearedVm() {
 
             break;
 
+        case RebootInProgress:
         case PoweringDown:
+            if (getVmManager().isColdReboot()) {
+                setColdRebootFlag();
+                resourceManager.removeAsyncRunningVm(dbVm.getId());
```

Comment:
Wondering what's the purpose of managing the "async running" VMs and what the VM removal from it here influences.

Reply:
If I understand correctly, if a VM is in `asyncRunningVms` list, it means that the VM was just started, began migration or resuming. If such a VM goes down, it will be rerun. Otherwise, if the VM is not in `asyncRunningVms` list, it means that the VM was expected to go down and it should not be rerun.

Reply:
I see, thank you for clarification.  Then it indeed has its purpose here.